identity/README.md

```diff
@@ -0,0 +1,279 @@
+  // the CSI-drivers's host company name and the CSI-driver name, to minimize
+  // the possibility of collisions. It MUST be 63 characters or less, beginning
+  // and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-),
+  // dots (.), and alphanumerics between. This field is REQUIRED.
+  string name = 1;
+  // This field is REQUIRED. Value of this field is opaque to the CO.
+  string vendor_version = 2;
+  // This field is OPTIONAL. Values are opaque to the CO.
+  map<string, string> manifest = 3;
+}
+```
+
+#### GetIdentity Errors
+
+If the CSI-driver is unable to complete the GetIdentity call successfully, it
+MUST return a non-ok gRPC code in the gRPC status.
+
+### `GetCapabilities`
+
+The CSI-Addons plugin for the CO needs to detect the supported features for
+each CSI-driver. Features are described as capabilities, each with additional
+information on how the CSI-Addons CO plugin may invoke the operation.
+
+There might be operations that need to run on the CSI Controller (provisioner)
+for storage system access. Other operations may require running on a volume
+that is attached/staged on a node. By specifying the type of Service that the
+CSI-driver provides, the CSI-Addons CO plugin can make informed decisions on
+where to invoke the RPCs for an operation.
+
+```protobuf
+// GetCapabilitiesRequest is sent by the CSI-Addons CO plugin to detect the
+// features that a CSI-driver supports.
+message GetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+// GetCapabilitiesResponse is returned by the CSI-driver as a response to a
+// GetCapabilitiesRequest.
+message GetCapabilitiesResponse {
+  // All the capabilities that the controller service supports. This
+  // field is OPTIONAL.
+  repeated Capability capabilities = 1;
+}
+
+// Specifies one or more capabilities of the CSI-driver.
+message Capability {
+  // Service contains the type of CSI Service that the CSI-driver provides.
+  message Service {
+    // Type describes a CSI Service that CSI-drivers can support.
+    enum Type {
+      // UNKNOWN indicates that the CSI-driver does not neither provide the CSI
+      // ControllerService or CSI NodeService. The CSI-Addons CO plugin will
+      // most likely ignore the node providing this Identify Service.
+      UNKNOWN = 0;
+
+      // CONTROLLER_SERVICE indicates that the CSI-driver provides RPCs for the
+      // CSI ControllerService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require access to the storage system,
+      // similar to the CSI Controller (provisioner).
+      CONTROLLER_SERVICE = 1;
+
+      // NODE_SERVICE indicates that the CSI-driver provides RPCs for the CSI
+      // NodeService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require a volume to be staged/attached to
+      // the node.
+      NODE_SERVICE = 2;
```

Comment:
if the plugin is registered as both Controller and node plugin(from both provisioner and node plugin) which one will take precedence?

Reply:
Indeed, it can be both. The CO-plugin will need to decide what operations to send to this CSI-driver. Some operations specified by CSI-Addons (like `ReclaimSpace`) may have operations for the CSI Controller and NodePlugin separately (storage cluster operation run on the CSI Controller, attached volume operations on the NodePlugin).